include/mfmg/dealii/lanczos.templates.hpp

```diff
@@ -66,10 +66,10 @@ Lanczos<OperatorType, VectorType>::solve(
   DeflatedOperator<OperatorType, VectorType> deflated_op(_op);
 
   // Loop over Lanczos solves
-  VectorType initial_guess(_op.n());
   for (int cycle = 0; cycle < num_cycles; ++cycle)
   {
-    details_set_initial_guess(initial_guess, cycle);
+    if (cycle > 0)
```

Comment:
I disagree with this. It would make sense not to rely on a caller to provide a randomly perturbed initial guess. I think `details_set_initial_guess` should still be called on the user provided guess.

Reply:
That's what I did at first but I fear than in three months we will want to use a particular initial guess and we will have forgotten that we don't use the user-provided initial guess. This will be a pain to debug. This is also consistent with what we do with arpack.

Reply:
Fair point, I have no counter argument.

Reply:
Add a comment to capture this discussion.